pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -266,6 +347,32 @@ func (l PipelineTaskList) Names() sets.String {
 	return names
 }
 
+// Validate a list of pipeline tasks including custom task and bundles
+func (l PipelineTaskList) Validate(ctx context.Context, taskNames sets.String, path string) (errs *apis.FieldError) {
+	for i, t := range l {
+		// validate pipeline task name
+		errs = errs.Also(t.ValidateName().ViaFieldIndex(path, i))
+		// names cannot be duplicated - checking that pipelineTask names are unique
+		if _, ok := taskNames[t.Name]; ok {
+			errs = errs.Also(apis.ErrMultipleOneOf("name").ViaFieldIndex(path, i))
+		}
+		taskNames.Insert(t.Name)
+		cfg := config.FromContextOrDefaults(ctx)
+		// If EnableCustomTasks feature flag is on, validate custom task specifications
+		// pipeline task having taskRef with APIVersion is classified as custom task
+		switch {
+		case cfg.FeatureFlags.EnableCustomTasks && t.TaskRef != nil && t.TaskRef.APIVersion != "":
+			errs = errs.Also(t.ValidateCustomTask().ViaFieldIndex(path, i))
+			// If EnableTektonOCIBundles feature flag is on, validate bundle specifications
+		case cfg.FeatureFlags.EnableTektonOCIBundles && t.TaskRef != nil && t.TaskRef.Bundle != "":
+			errs = errs.Also(t.ValidateBundle().ViaFieldIndex(path, i))
+		default:
+			errs = errs.Also(t.Validate(ctx).ViaFieldIndex(path, i))
```

Comment:
Any reason for doing this in the `PipelineTaskList.Validate` function instead of in the `PipelineTask.Validate` ?

Reply:
I moved `switch` statement to `PipelineTask.Validate`, wdyt?